CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -777,6 +777,12 @@ public static function recurringContribution(&$form) {
     CRM_Core_Form_Date::buildDateRange($form, 'contribution_recur_next_sched_contribution_date', 1, '_low', '_high', ts('From'), FALSE, FALSE, 'birth');
     CRM_Core_Form_Date::buildDateRange($form, 'contribution_recur_failure_retry_date', 1, '_low', '_high', ts('From'), FALSE, FALSE, 'birth');
     CRM_Core_Form_Date::buildDateRange($form, 'contribution_recur_cancel_date', 1, '_low', '_high', ts('From'), FALSE, FALSE, 'birth');
+
+    // Add field for contribution status
+    $form->addSelect('contribution_recur_contribution_status_id',
+      array('entity' => 'contribution', 'multiple' => 'multiple', 'context' => 'search', 'options' => CRM_Contribute_PseudoConstant::contributionStatus()),
```

Comment:
Oops you left a comma ```...CRM_Contribute_PseudoConstant::contributionStatus()),``` at then end. Causes test failure again :(

Reply:
Sounds like I need to install PHP code sniffer after all